### DISCUSSION TITLE
What on earth is NueJS

### DISCUSSION THREAD

**Discussion by jcs021102** [nuejs/nue#333] [Q&A]
Hi, sorry in advance if I come off as rude, but im honestly quite frustrated right now. I am writing this as a question, and criticism, as nothing that claims to be UX oriented, should illicit this response from a user. 

I just stumbled over NueJS, marketed as a `Web framework for UX developers` but after a good 30minutes of looking into it, I still have no clue what makes it so special.

The `What is Nue?` page gives me no information whatsoever. It has benchmarks, it shows screenshots of figma yet im given no indication of how it's integrated. I shouldn't have to go through more than a brief skim of a page to get a good understanding of what the product offers. yet I've had to go through a couple videos and an about page to be left completely uninformed.

So TLDR:

Question, what actually is nuejs and how would it makes my UX Design experience better as a UX oriented developer.

Criticism: The answer to the above question should be as blatantly obvious as possible on the homepage and about page. The fact that it does not do so, is insulting towards UX professionals.
Also maybe create a discord or other form of communication aside from github discussions

**Comment by kyoukhana** [comment]
I understand your frustration, but keep in mind that Nue is still in its development stages. It's a much better solution than Astro, very lightweight and with fewer dependencies. The basic Nu blog is clean. Give it time, and I'm sure things will become more clear for you. Many developers are burned out from the bloat of front-end frameworks, and honestly, we don't need them anymore. We can return to the basics of just HTML and CSS with what ever back-end you need. While Tailwind is a solution, it can become bloated over time. The developer is aiming to simplify the development process and make it easier to build something.

**Reply by tomByrer** [reply]
Well, Astro is great if you have to support other frameworks like React, Svelte, Vue, Solid... or all 4 at once!  & Astro community is centered around Tailwind; a plus for some but negative for me.

**Comment by tipiirai** [comment]
Hey @jcs021102 , 

To answer your question we need to understand the important difference between a UX developers and the JS engineers. Current frameworks like Next.js and Vite are targeted for JS engineers, whose interests, responsibilities, and skillsets are heavily revolved around JavaScript. Nue is targeted for UX developers, whose interests, responsibilities, and skillsets are focused on other areas of the front end, like HTML, CSS, design, interaction, patterns, accessibility, etc. Check out these great articles for the division:

https://css-tricks.com/the-great-divide/
https://bradfrost.com/blog/post/front-of-the-front-end-and-back-of-the-front-end-web-development/

Basically the difference is that your codebase is mostly CSS instead of TypeScript. And you need significanly less code to do the same thing — given that you are fluent with CSS. 

> insulting towards UX professionals.

Maybe your notion about UX professional is something different? I'm very curious to hear about that. Might help improve the docs.

**Reply by tomByrer** [reply]
correction:
your codebase is mostly CSS _and HTML_ instead of TypeScript _& JSX_

My elevator pitch would be,
"NueJS takes only a few hours to be productive with if you know HTML, CSS, & basic JS, versus other frameworks that may take days or weeks."

**Comment by tipiirai** [comment]
@jcs021102 — now I actually get what you mean. There's a lot room for improvement on the introduction. Will push out something soon. Will also give a good preparation and structure for my talk on [JS Nation](https://jsnation.us/#speakers)

**Reply by nobkd** [reply]
@tipiirai 
Note: The github link on your gitnation page (https://gitnation.com/person/tero_piirainen) wrongly links to gh/nue instead of gh/nuejs or gh/nuejs/nue

**Reply by tipiirai** [reply]
Just sent them email about this. Thanks for the reminder!

**Comment by jcs021102** [comment]
Do update me when this is done, as I still am interested in what your
service offers.

**Reply by tipiirai** [reply]
Deal

**Reply by tipiirai** [reply]
@jcs021102 the update is now done. Check out details:

https://nuejs.org/blog/nue-release-candidate/

Please let me know how it looks like to you now. Is Nue still confusing? Thanks in advance.

**Comment by jcs021102** [comment]
First line already did more than the entire old page. Overall I think it
looks fantastic, i'm actually quite excited for it, it's giving me some
astro vibes as a first impression but I quite like the direction.
I'd look over the benchmarking section a bit, the 5 × smaller wording is a
bit odd, but aside from that I'm chuffed to bits. Do yall have a discord?
I'd love to see where this project goes moving forward, hey I might even
contribute down the line. :)

**Reply by tipiirai** [reply]
Hey! Great to hear. You mean the "Can it outshine Astro?" line or "The Design Engineering Framework" line? I'm still struggling a bit with the headline, since "Design Engineering", while a fairly common term, it might sound strange to people. Maybe it should be along the lines of "The simple web framework" or "The simple web stack" or even "The standard web stack". 

Where is the 5 x smaller thing? Your're right — that should be taken away. 

There is no Discord. There is a small Slack group.

**Comment by jcs021102** [comment]
I'd consider discord as a forum for anyone interested in chatting, nothing
wrong with slack either

**Reply by tipiirai** [reply]
You can join [here](https://join.slack.com/t/nuejs/shared_invite/zt-2ufdia6ty-EPEgUK1Un8q4sYDMB8DbdA)
